src/librustdoc/lib.rs

```diff
@@ -269,6 +269,26 @@ fn opts() -> Vec<RustcOptGroup> {
                 "sort modules by where they appear in the program, rather than alphabetically",
             )
         }),
+        unstable("default-theme", |o| {
+            o.optopt(
+                "",
+                "default-theme",
+                "Set the default theme. THEME should be the theme name, generally lowercase. \
+                 If an unknown default theme is specified, the builtin default is used. \
+                 The set of themes, and the rustdoc built-in default is not stable.",
+                "THEME",
+            )
+        }),
```

Comment:
Why is default-theme special cased? If you want `default-settings` to work for arbitrary settings, I think it makes sense to use it for themes too. If you want this to be documented, I would rather document settings in the unstable section of the rustdoc book (`src/doc/rustdoc/src/unstable-features.md`).

Reply:
(I would also be ok with getting rid of `default-settings` and only keeping `default-themes`.)

Reply:
The storage names are rather internal, and `--default-setting` sets them directly.  That's good for flexibility but it is rather too raw.

Indeed, as you can see from the implementation, `--default-theme` doesn't just set the `theme` setting.  Because of the way the "use system theme" option and the light/dark themes operate, since #77809, it is necessary to change the default for "use system theme" to "false" for the `--default-theme` option to work.

Indeed, I think this fact about #77809 shows that my pre-#77809 approach of having a raw `--default-setting` option, and a cooked `--default-theme` option, was correct.  Before #77809, `--default-theme` just set the `theme` setting.  Now it does more.

Notice that the stability caveats for the two options are rather different.  Another aspect of this is that it might easily make sense to stabilise the `--default-theme` option soon, after we've had some experience with it.  Stabilising the raw `--default-setting` option would involve thinking about what additional promises that would imply (maybe none that we care about? but I'm not sure and it's certainly harder to think about)